services/f-droid/f-droid.service.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
+
+const BaseHTTPService = require('../base-http')
+const { addv: versionText } = require('../../lib/text-formatters')
+const { version: versionColor } = require('../../lib/color-formatters')
+const { InvalidResponse } = require('../errors')
+
+module.exports = class FDroid extends BaseHTTPService {
+  async fetch({ appId }) {
+    const url = `https://f-droid.org/en/packages/${appId}/`
+    return this._requestHTTP({
+      url,
+      options: {},
+      errorMessages: {
+        404: 'app not found',
+      },
+    }).then(({ res, buffer }) => {
+      const website = buffer.toString()
+      // we assume the layout as provided here:
+      // https://gitlab.com/fdroid/fdroid-website/blob/9ae61894a18889ed749d36d5afbd0db3d0b0cfdd/_layouts/package.html#L147
+      const match = website.match(
+        /<div\s[^>]*class="package-version-header"(?:\s[^>]*)?>[^<]*<a\s+name="([^:>]*)"(?:\s[^>]*)?>/
+      )
+      if (!match) {
+        throw new InvalidResponse({
+          prettyMessage: 'fix this badge',
+          underlyingError: new Error('could not find version on website'),
+        })
```

Comment:
Do we have a customary message we use here? **package not found**?

Though I do like how direct **fix this badge** is…

Reply:
Ah, I understand what this is doing now. It's not the not found message; it's the message that appears if the web page changes.


If this case still exists after the data source update, could you update this message so it reads `'invalid response'`?